docs2/site/docs/migrations/migration4.md

```diff
@@ -59,40 +103,93 @@
-  to create a copy that is not re-used. Note that it is safe to use the field context within asynchronous field resolvers and
-  data loaders. Once the asynchronous field resolver or data loader returns its final result, the context may be re-used.
-  Also, any calls to the configured `UnhandledExceptionDelegate` will receive a field context copy that will not be re-used,
-  so it is safe to preserve these instances without calling `.Copy()`.
 * `ExecutionHelper.CollectFields` method was moved into `Fields` class and renamed to `CollectFrom`
-* `IProvideMetadata.Metadata` is now `Dictionary` instead of `ConcurrentDictionary`, and is not thread safe anymore
-* By default, descriptions for fields, types, enums, and so on are not pulled from xml comments unless the corresponding global flag is enabled.
 * `ISchema.FindDirective`, `ISchema.RegisterDirective`, `ISchema.RegisterDirectives` methods were moved into `SchemaDirectives` class
 * `ISchema.FindType` method was moved into `SchemaTypes[string typeName]` indexer
 * Some of the `ISchemaNodeVisitor` methods have been changes to better support schema traversal
+* `SourceLocation`, `NameNode` and `BasicVisitor` have been changed to a `readonly struct`.
+* `ObjectExtensions.GetInterface` has been removed along with two overloads of `GetPropertyValue`.
+* `void Visit<TState>(System.Action<INode, TState> action, TState state)` method has been added.
+* Various `IEnumerable<T>` properties on schema and graph types have been changed to custom collections: `SchemaDirectives`, `SchemaTypes`, `TypeFields`, `PossibleTypes`, `ResolvedInterfaces`
+* `INode.IsEqualTo` and related methods have been removed.
+* `ApolloTracing.ConvertTime` is now private and `ResolverTrace.Path` does not initialize an empty list when created.
+
+### `ExecutionOptions.EnableMetrics` is disabled by default
+
+To enable metrics, please set the option to `true` before executing the query.
+
+(todo: add sample)
+
+### GraphQL Member Descriptions
+
+> By default, descriptions for fields, types, enums, and so on are not pulled from xml comments unless the corresponding global flag is enabled.
+
+Note that to improve performance, by default GraphQL.NET 4.0 does not pull descriptions from fields from xml comments as it did in 3.x. To re-enable that functionality, see Global Switches above.
+
+(todo: update/consolidate description)
+
+### Changes to `IResolveFieldContext.Arguments`
+
+`IResolveFieldContext.Arguments` now returns an `IDictionary<string, ArgumentValue>` instead of `IDictionary<string, object>` so that it
+can be determined if the value returned is a default value or if it is a specified literal or variable.
+
+`IResolveFieldContext.HasArgument` now returns `false` when `GetArgument` returns a field default value. Note that if a variable is specified,
+and the variable resolves to its default value, then `HasArgument` returns `true` (since the field argument is successfully resolving to a variable
```

Comment:
```suggestion
and the variable resolves to its default value, then `HasArgument` returns `true` (since the field argument is successfully resolved to a variable
```

?